torchaudio/functional.py

```diff
@@ -1287,10 +1288,7 @@ def overdrive(
     output_waveform = torch.zeros_like(waveform, dtype=dtype, device=device)
 
     # TODO: Implement a torch CPP extension
-    for i in range(waveform.shape[-1]):
-        last_out = temp[:, i] - last_in + 0.995 * last_out
-        last_in = temp[:, i]
```

Comment:
This is strange. last_in will be temp[:, i - 1] after the first iteration if I'm not mistaken. Why always do the copy?